src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,36 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
-
+    
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new ValidationException("Any user value cannot be null");
+        }
+        
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("User with this login exists try again");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new ValidationException("Login is too short, it must be more than 6 characters");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new ValidationException(
+                    "Password is too short, it must be more than 6 characters");
+        }
+        if (user.getAge() < MIN_USER_AGE) {
+            throw new ValidationException("User must be over 18 years of age");
+        }
```

Comment:
Hard to read.
It is better to divide into methods and describe the necessary logic in them.
For example, you can create private methods:
```
private checkAge(User user) {
// age verification
}
```